Intersect.Client/Entities/Player.cs

```diff
@@ -787,63 +846,99 @@ private void WithdrawItemInputBoxOkay(object sender, EventArgs e)
-                        Inventory[invSlot].Quantity
-                    );
-                }
-                else
-                {
-                    PacketSender.SendStoreBagItem(invSlot, 1, bagSlot);
-                }
+            if (inventorySlot.Quantity < 2)
+            {
+                PacketSender.SendStoreBagItem(inventorySlotIndex, 1, bagSlotIndex);
+                return;
             }
+
+            int[] userData = new int[2] { inventorySlotIndex, bagSlotIndex };
+
+            var quantity = inventorySlot.Quantity;
+            var maxQuantity = quantity;
+
+            // TODO: Refactor server bagging logic to allow transferring quantities from the entire inventory and not just the slot
+            //if (itemDescriptor.IsStackable)
+            //{
+            //    maxQuantity = GetQuantityOfItemInInventory(itemDescriptor.Id);
+            //}
+
+            InputBox.Open(
+                title: Strings.Bags.storeitem,
+                prompt: Strings.Bags.storeitemprompt.ToString(itemDescriptor.Name), true,
+                inputType: InputBox.InputType.NumericSliderInput,
+                onSuccess: StoreBagItemInputBoxOkay,
+                onCancel: null,
+                userData: userData,
+                quantity: quantity,
+                maxQuantity: maxQuantity
+            );
         }
 
         private void StoreBagItemInputBoxOkay(object sender, EventArgs e)
         {
-            var value = (int)((InputBox)sender).Value;
+            var value = (int)Math.Round(((InputBox)sender).Value);
             if (value > 0)
             {
                 int[] userData = (int[])((InputBox)sender).UserData;
                 PacketSender.SendStoreBagItem(userData[0], value, userData[1]);
             }
         }
 
-        public void TryRetreiveBagItem(int bagSlot, int invSlot)
+        public void TryRetreiveBagItem(int bagSlotIndex, int inventorySlotIndex)
         {
-            if (Globals.Bag[bagSlot] != null && ItemBase.Get(Globals.Bag[bagSlot].ItemId) != null)
+            var bagSlot = Globals.Bag[bagSlotIndex];
+            if (bagSlot == default)
             {
-                int[] userData = new int[2] { bagSlot, invSlot };
+                return;
+            }
 
-                if (Globals.Bag[bagSlot].Quantity > 1)
-                {
-                    var iBox = new InputBox(
-                        Strings.Bags.retreiveitem,
-                        Strings.Bags.retreiveitemprompt.ToString(ItemBase.Get(Globals.Bag[bagSlot].ItemId).Name), true,
-                        InputBox.InputType.NumericSliderInput, RetreiveBagItemInputBoxOkay, null, userData, Globals.Bag[bagSlot].Quantity,
-                        Globals.Bag[bagSlot].Quantity
-                    );
-                }
-                else
-                {
-                    PacketSender.SendRetrieveBagItem(bagSlot, 1, invSlot);
-                }
+            if (bagSlot.Quantity < 2)
```

Comment:
So I moved it because since bagging can't store/retrieve more than one slot at a time the maxQuantity check wasn't needed so it could be put earlier for performance, but I ended up fixing it because I don't know if I really changed it for performance or just forgot it. Rather than adding a comment I decided fixing it now while it's fresh is better.